### PR TITLE
Add email field to user profile

### PR DESCRIPTION
**Pull request by JenySadadia**
Add `UserProfile.email` field to store user's email address. Update POST and PUT user handlers accordingly.
Also, update script to setup admin user to get admin email address.

### CODE REVIEW

File: api/admin.py [hunk 1]
```diff
@@ -65,5 +66,7 @@ async def main(args):
                         help="Admin group name")
     parser.add_argument('--database', default='kernelci',
                         help="KernelCI database name")
+    parser.add_argument('email',
+                        help="Admin user email address")
```
Comment (gctucker) [inline api/admin.py:70]: All the other args are non-positional so for consistency:
```suggestion
    parser.add_argument('--email', required=True,
                        help="Admin user email address")
```

File: api/models.py [hunk 2]
```diff
@@ -22,6 +22,7 @@
     FileUrl,
     SecretStr,
     conlist,
+    EmailStr,
```
Comment (gctucker) [inline api/models.py:25]: It might be a bit better to preserve the alphabetical order of the imports.

File: api/models.py [hunk 3]
```diff
@@ -125,6 +126,7 @@ class UserProfile(BaseModel):
         default=[],
         description="A list of groups that user belongs to"
     )
+    email: EmailStr
```
Comment (gctucker) [inline api/models.py:129]: Please define it as a `Field` with a description so that appears in the auto-generated documentation.

File: api/main.py [hunk 4]
```diff
@@ -141,6 +141,7 @@ async def authorize_user(node_id: str, user: User = Depends(get_current_user)):
           response_model_by_alias=False)
 async def post_user(
         username: str, password: Password,
+        email: str,
```
Comment (gctucker) [inline api/main.py:144]: Just for readability:
```suggestion
        username: str, password: Password, email: str,
```

**Review by gctucker — COMMENTED** [review]
A few minor things, LGTM otherwise.

**Comment by JenySadadia** [comment]
Thanks for the review @gctucker. Addressed all the comments.

**Comment by gctucker** [comment]
Please let me know when you've updated the user documents on staging and I'll propagate it to the Atlas database.

**Comment by JenySadadia** [comment]
> Please let me know when you've updated the user documents on staging and I'll propagate it to the Atlas database.

What should be the email address of the admin user on staging?

**Comment by JenySadadia** [comment]
Added `email` to all staging users.

**Comment by gctucker** [comment]
OK great, thanks.  I've now updated the user collection in Atlas:
```
kernelci@kernelci-staging:~/dump$ mongodump --host localhost --port 8017 --db kernelci -c user
2023-07-28T17:45:48.976+0000	writing kernelci.user to dump/kernelci/user.bson
2023-07-28T17:45:48.982+0000	done dumping kernelci.user (5 documents)
kernelci@kernelci-staging:~/dump$ mongo mongodb+srv://kernelci-staging-api:<password>@kernelci-api-staging-az.vbocidn.mongodb.net
# ...
MongoDB Enterprise atlas-ucvcf2-shard-0:PRIMARY> db.user.remove({})
WriteResult({ "nRemoved" : 9 })
# ...
kernelci@kernelci-staging:~/dump$ mongorestore --uri mongodb+srv://kernelci-staging-api:<password>@kernelci-api-staging-az.vbocidn.mongodb.net --db kernelci -c user dump/kernelci/user.bson 
2023-07-28T17:48:16.142+0000	checking for collection data in dump/kernelci/user.bson
2023-07-28T17:48:16.188+0000	restoring to existing collection kernelci.user without dropping
2023-07-28T17:48:16.188+0000	reading metadata for kernelci.user from dump/kernelci/user.metadata.json
2023-07-28T17:48:16.188+0000	restoring kernelci.user from dump/kernelci/user.bson
2023-07-28T17:48:16.268+0000	restoring indexes for collection kernelci.user from metadata
2023-07-28T17:48:16.357+0000	finished restoring kernelci.user (5 documents, 0 failures)
2023-07-28T17:48:16.357+0000	5 document(s) restored successfully. 0 document(s) failed to restore.
```

**Comment by JenySadadia** [comment]
Email field is visible here:
```
$ curl https://staging.kernelci.org:9000/latest/users/profile
{"items":[{"profile":{"username":"jeny","groups":[],"email":"jeny.sadadia@collabora.com"}},{"profile":{"username":"staging.kernelci.org","groups":[],"email":"kernelci-tsc@groups.io"}},{"profile":{"username":"gtucker","groups":[],"email":"guillaume.tucker@collabora.com"}},{"profile":{"username":"rcn","groups":[],"email":"ricardo.canuelo@collabora.com"}},{"profile":{"username":"admin","groups":[{"id":"649190b01432ccfdb575a250","name":"admin"}],"email":"kernelci-tsc@groups.io"}}],"total":5,"limit":50,"offset":0
```

File: tests/e2e_tests/test_user_creation.py [hunk 5]
```diff
@@ -38,6 +38,7 @@ async def test_create_admin_user(test_async_client):
     profile = UserProfile(
         username=username,
         hashed_password=hashed_password,
+        email='test-admin@gmail.com',
```
Comment (gctucker) [inline tests/e2e_tests/test_user_creation.py:41]: It might be better to use `@kernelci.org` email addresses, just in case some code runs this and actually tries to send emails to some GMail address that might really exist.  But that's really a corner-case thing.

**Review by gctucker — APPROVED** [review]
Just a minor thing, I think we need to be a bit careful when using things like URLs and emails in test code in case it actually resolves as something that exists.  Also, we might in the future have an actual KernelCI test email address for complete end-to-end testing.  If it's too much hassle to change this now or it doesn't seem worth it then I think it can be merged as-is.

File: tests/e2e_tests/test_user_creation.py (hunk 5 above)
Comment (JenySadadia) [reply tests/e2e_tests/test_user_creation.py:41]: Okay. Yes, I agree with your point. I'll update this and other tests as well. Thanks.

**Review by gctucker — APPROVED** [review]
